src/snowflake/cli/api/sql_execution.py

```diff
@@ -82,44 +82,27 @@ def use_role(self, new_role: str):
             if is_different_role:
                 self._execute_query(f"use role {prev_role}")
 
-    def _execute_schema_query(self, query: str, **kwargs):
-        self.check_database_and_schema()
-        return self._execute_query(query, **kwargs)
-
-    def check_database_and_schema(self) -> None:
+    def _execute_schema_query(self, query: str, name: Optional[str] = None, **kwargs):
         """
-        Checks if the connection database and schema are set and that they actually exist in Snowflake.
+        Check that a database and schema are provided before executing the query. Useful for operating on schema level objects.
         """
-        self.check_schema_exists(self._conn.database, self._conn.schema)
+        self.check_database_and_schema_provided(name)
+        return self._execute_query(query, **kwargs)
 
-    def check_database_exists(self, database: str) -> None:
+    def check_database_and_schema_provided(self, name: Optional[str] = None) -> None:
         """
-        Checks that database is provided and that it is a valid database in
-        Snowflake. Note that this could fail for a variety of reasons,
-        including not authorized to use database, database doesn't exist,
-        database is not a valid identifier, and more.
+        Checks if a database and schema are provided, either through the connection context or a qualified name.
         """
+        if name:
+            _, schema, database = from_qualified_name(name)
+        else:
+            schema, database = None, None
+        schema = schema or self._conn.schema
+        database = database or self._conn.database
         if not database:
             raise DatabaseNotProvidedError()
-        try:
-            self._execute_query(f"USE DATABASE {database}")
-        except ProgrammingError as e:
```

Comment:
Note that I removed the execution of USE DATABASE and USE SCHEMA in use_schema_query. I think that it makes sense to check a database and schema are provided but if they are necessary for the query then an error will be raised when executing the main query if the schema or database do not actually exist.